### PR TITLE
Omarbr

### PR DESCRIPTION
**Pull request by Omarmanassrah**
This Branch contain the solution of the python task that include Initialize a list and filter it and use the match case.

### CODE REVIEW

File: Main.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# FIRST-NAME_LAST-NAME.py
+# This file is a template and only an example. Please customize as needed.
+
+
+# Step 1: Initialize Book Inventory
```
Comment (Mohammed-samaheen) [inline Main.py:5]: These comments are meant to be placed before your changes. Move these comments to the expected code block.

**Review by Mohammed-samaheen — CHANGES_REQUESTED** [review]

File: Main.py (hunk 1 above)
Comment (Omarmanassrah) [reply Main.py:5]: I add it right now

File: FIRST-NAME_LAST-NAME.py
Comment (FaresAbuIram) [inline FIRST-NAME_LAST-NAME.py:1]: Please don't remove this file

File: Main.py
Comment (FaresAbuIram) [inline Main.py:1]: Please change the  file name to your name 
FirstName_LastName.py

File: Main.py (hunk 1 above)
Comment (FaresAbuIram) [reply Main.py:5]: @Omarmanassrah 
Please move the right comments before each step, example 
Step 1: Initialize Book Inventory 
Here write your code 
`books= [...]`

Step 2: ....

**Review by FaresAbuIram — CHANGES_REQUESTED** [review]

**Comment by FaresAbuIram** [comment]
Don't forget to add a good description

File: Omar_Manassra.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+
+# Step 1: Initialize Book Inventory
+
+books = [
+    {"Title": "Python Fundamentals", "Author": "Ahmed Zayed", "Availability": True, "Tags": ["Programming", "Python", "Basics"]},
+    {"Title": "The Art of Coding", "Author": "Mohammed Ali", "Availability": False, "Tags": ["Coding", "Art", "Creativity"]},
+    {"Title": "Data Structures in Python", "Author": "Yousef Amr", "Availability": True, "Tags": ["Data", "Python", "Intermediate"]}
+]
+# Step 2: Filter Books by Author and Availability
```
Comment (FaresAbuIram) [inline Omar_Manassra.py:26]: please add an empty line before this line.

File: Omar_Manassra.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+# This file is a template and only an example. Please customize as needed.
+for book in books:
+    match book["Tags"][0]:
+        case "Programming":
+            print(f'The book "{book["Title"]}" is related to programming.')
+        case "Coding":
+            print(f'The book "{book["Title"]}" is related to coding.')
+        case "Data":
+            print(f'The book "{book["Title"]}" is related to data.')
+
```
Comment (FaresAbuIram) [inline Omar_Manassra.py:48]: we still need another case If none of these tags [Programming, Coding, or Data] are present, print the book's title with a message indicating that the tag is unrecognized.

**Comment by Omarmanassrah** [comment]
I resolve everything if there is anything to change please tell me thank you very much for your efforts

File: FIRST-NAME_LAST-NAME.py [hunk 4]
```diff
@@ -12,5 +12,4 @@
 
 
 # Step 3: Analyze Tags with Match Statement
-# For each book, use a match statement to check the first tag and print relevant messages.
-
+# For each book, use a match statement to check the first tag and print relevant messages.
```
Comment (FaresAbuIram) [inline FIRST-NAME_LAST-NAME.py:15]: the last change, please add an empty line after this line 

We don't need to see the whole file here. In the file changed section, we only need to see one file, which is Omar_Manassra.py. just add an empty line

Comment (Omarmanassrah) [reply FIRST-NAME_LAST-NAME.py:15]: the empty line is 15 and put the last line at 16 or to add an empty line after them at 16

Comment (FaresAbuIram) [reply FIRST-NAME_LAST-NAME.py:15]: the line 15 is good, but you need to add line 16 
which is an empty file 

in your changes, you removed line #16

Comment (FaresAbuIram) [reply FIRST-NAME_LAST-NAME.py:15]: Screenshot 2024-08-11 at 15 32 15

Comment (FaresAbuIram) [reply FIRST-NAME_LAST-NAME.py:15]: Screenshot 2024-08-14 at 11 45 27

Comment (Omarmanassrah) [reply FIRST-NAME_LAST-NAME.py]: I do it

File: FIRST-NAME_LAST-NAME.py [hunk 5]
```diff
@@ -13,4 +13,3 @@
 
 # Step 3: Analyze Tags with Match Statement
 # For each book, use a match statement to check the first tag and print relevant messages.
-
```
Comment (FaresAbuIram) [inline FIRST-NAME_LAST-NAME.py:16]: red means this line removed 
just hit enter after line 15

Comment (Omarmanassrah) [reply FIRST-NAME_LAST-NAME.py:16]: what about now i hit enter

Comment (FaresAbuIram) [reply FIRST-NAME_LAST-NAME.py:16]: I can't see that

Comment (FaresAbuIram) [reply FIRST-NAME_LAST-NAME.py:16]: try to do it again

Comment (Omarmanassrah) [reply FIRST-NAME_LAST-NAME.py:16]: i send a picture for you at slack

Comment (Omarmanassrah) [reply FIRST-NAME_LAST-NAME.py:16]: I repeat it again also

File: Main.py (hunk 1 above)
Comment (Omarmanassrah) [reply Main.py]: I resolve it @Mohammed-samaheen are you approved to it

**Review by Mohammed-samaheen — APPROVED** [review]

**Comment by Mohammed-samaheen** [comment]
LGTM

**Pull request merged**